### ISSUE TITLE
Java, version 5.3.4 java.lang.UnsatisfiedLinkError

### ISSUE THREAD

**Issue by ccleve** [facebook/rocksdb#2281]
```
Exception in thread "main" java.lang.UnsatisfiedLinkError: org.rocksdb.Options.newOptions()J
	at org.rocksdb.Options.newOptions(Native Method)
	at org.rocksdb.Options.<init>(Options.java:31)
	at com.dieselpoint.standardkv.impl.rocksdb.Test.test1(Test.java:36)
	at com.dieselpoint.standardkv.impl.rocksdb.Test.main(Test.java:28)
```
fails on this line: `Options options = new Options();`

The Options class in the jar appears to be empty.
Happens in version 5.3.4 on Windows.

@adamretter

**Comment by ccleve** [comment]
The Options class is actually not empty (IDE glitch). But the error is still there.

**Comment by ccleve** [comment]
I'm on Windows

**Comment by adamretter** [comment]
@ccleve Okay I can see the problem and reproduce it here

If I run `dumpbin /exports librocksdbjni-win64.dll` against the DLL from RocksDB 5.1.4, then I see all of the `Java_` exported functions that I would expect. However if I run it against a DLL from RocksDB 5.3.4 then I don't see any `Java_` JNI exported functions, just the RocksDB C++ API exported functions.

I am afraid to report that it looks like it is possible I made a mistake when preparing the Windows build (which is a very manual process). I am not certain, but it looks likely that I mistakenly copied `build\Release\rocksdb-shared.dll` into the Jar packages and not `build/java/Release/rocksdbjni.dll`.

It looks like I will need to produce another release ASAP anyway, as we had an issue on Linux too - https://github.com/facebook/rocksdb/issues/2280.

@ccleve I did make some progress building a better Windows binary as documented here - https://github.com/facebook/rocksdb/wiki/Building-on-Windows but then hit an issue here - https://github.com/facebook/rocksdb/issues/2293

**Comment by yuslepukhin** [comment]
@adamretter You may want to check if your entry points are marked __dllexport when the code is compiled  with DLL target in mind. See the example in the c.h where the C interfaces are exported.

**Comment by adamretter** [comment]
@ccleve We just released 5.3.6. I believe that should fix your issue.

**Comment by ccleve** [comment]
Confirmed: it does fix the issue. Thank you.
